File: golang/tests/pkg/integration/get_session_integration_test.go
package integration

import (
	"fmt"
	"os"
	"testing"

	"github.com/aliyun/wuying-agentbay-sdk/golang/pkg/agentbay"
)

// TestGetStatusAPI integration test for GetSessionDetail (GetStatus) API
func TestGetStatusAPI(t *testing.T) {
	// Get API Key
	apiKey := os.Getenv("AGENTBAY_API_KEY")
	if apiKey == "" {
		t.Fatal("AGENTBAY_API_KEY environment variable is not set")
	}

	// Initialize AgentBay client
	client, err := agentbay.NewAgentBay(apiKey, nil)
	if err != nil {
		t.Fatalf("Failed to initialize AgentBay client: %v", err)
	}

	// Create a session first
	fmt.Println("Creating a new session for GetSession testing...")
	createResult, err := client.Create(nil)
	if err != nil {
		t.Fatalf("Failed to create session: %v", err)
	}
	session := createResult.Session
	sessionID := session.SessionID
	t.Logf("Session created with ID: %s", sessionID)

	// Test GetStatus API
	fmt.Println("Testing GetStatus API...")
	getStatusResult, err := session.GetStatus()
	if err != nil {
		t.Fatalf("Failed to call GetStatus API: %v", err)
	}

	// Validate response
	if getStatusResult.RequestID == "" {
		t.Error("RequestID should not be empty")
	}
	t.Logf("GetStatus RequestID: %s", getStatusResult.RequestID)

	if getStatusResult.HttpStatusCode != 200 {
		t.Errorf("Expected HttpStatusCode 200, got %d", getStatusResult.HttpStatusCode)
	}

	if getStatusResult.Code != "ok" {
		t.Errorf("Expected Code 'ok', got '%s'", getStatusResult.Code)
	}

	if !getStatusResult.Success {
		t.Error("Expected Success to be true")
	}

	if getStatusResult.Status == "" {
		t.Error("Status should not be empty")
	}
	t.Logf("Status: %s", getStatusResult.Status)

	// TODO: Don't support file transfer context yet
	// Validate contexts field
	// if getSessionResult.Data.Contexts == nil {
	// 	t.Error("Contexts field should not be nil")
	// }
	// if len(getSessionResult.Data.Contexts) == 0 {
	// 	t.Error("Contexts list should not be empty")
	// }
	// t.Logf("Contexts count: %d", len(getSessionResult.Data.Contexts))
	// for i, ctx := range getSessionResult.Data.Contexts {
	// 	if ctx.Name == "" {
	// 		t.Errorf("Context %d should have a non-empty Name field", i)
	// 	}
	// 	if ctx.ID == "" {
	// 		t.Errorf("Context %d should have a non-empty ID field", i)
	// 	}
	// 	t.Logf("Context %d: name=%s, id=%s", i, ctx.Name, ctx.ID)
	// }

	// Clean up: Delete the session
	fmt.Println("Cleaning up: Deleting the session...")
	_, err = session.Delete()
	if err != nil {
		t.Logf("Warning: Failed to delete session: %v", err)
	} else {
		t.Logf("Session %s deleted successfully", sessionID)
	}
}

File: typescript/tests/integration/get-session.integration.test.ts
/**
 * Integration test for GetSession API
 */

import { AgentBay } from "../../src/agent-bay";
import { Session } from "../../src/session";
import { log } from "../../src/utils/logger";

describe('GetSession API Integration Test', () => {
  let agentBay: AgentBay;
  const apiKey = process.env.AGENTBAY_API_KEY;

  beforeAll(() => {
    if (!apiKey) {
      throw new Error('AGENTBAY_API_KEY environment variable is not set');
    }
    agentBay = new AgentBay({ apiKey });
  });

  it('should retrieve session information using GetSession API', async () => {
    // Create a session first
    log('Creating a new session for GetSession testing...');
    const createResult = await agentBay.create({ imageId: "linux_latest"});
    expect(createResult.success).toBe(true);
    expect(createResult.session).toBeDefined();

    const session = createResult.session!;
    const sessionId = session.sessionId;
    log(`Session created with ID: ${sessionId}`);

    try {
      // Test GetSession API
      log('Testing GetSession API...');
      const getSessionResult = await agentBay.getSession(sessionId);
      // Validate response
      expect(getSessionResult.requestId).toBeTruthy();
      log(`GetSession RequestID: ${getSessionResult.requestId}`);

      expect(getSessionResult.httpStatusCode).toBe(200);
      expect(getSessionResult.code).toBe('ok');
      expect(getSessionResult.success).toBe(true);

      // Validate Data field
      expect(getSessionResult.data).toBeDefined();
      expect(getSessionResult.data!.sessionId).toBe(sessionId);
      // Note: data.success may reflect a different status than the overall response
      // We validate the overall response success instead
      expect(getSessionResult.data!.appInstanceId).toBeTruthy();
      log(`AppInstanceID: ${getSessionResult.data!.appInstanceId}`);
      expect(getSessionResult.data!.resourceId).toBeTruthy();
      log(`ResourceID: ${getSessionResult.data!.resourceId}`);
      log('GetSession API test passed successfully');
    } finally {
      // Clean up: Delete the session
      log('Cleaning up: Deleting the session...');
      const deleteResult = await session.delete();
      if (deleteResult.success) {
        log(`Session ${sessionId} deleted successfully`);
      } else {
        log(`Warning: Failed to delete session: ${deleteResult.errorMessage}`);
      }
    }
  });
});

